internal/email/email.go

```diff
@@ -35,13 +35,14 @@ type email struct {
 
 // Config email agent configuration.
 type Config struct {
-	Host        string `env:"SMQ_EMAIL_HOST"         envDefault:"localhost"`
-	Port        string `env:"SMQ_EMAIL_PORT"         envDefault:"25"`
-	Username    string `env:"SMQ_EMAIL_USERNAME"     envDefault:"root"`
-	Password    string `env:"SMQ_EMAIL_PASSWORD"     envDefault:""`
-	FromAddress string `env:"SMQ_EMAIL_FROM_ADDRESS" envDefault:""`
-	FromName    string `env:"SMQ_EMAIL_FROM_NAME"    envDefault:""`
-	Template    string `env:"SMQ_EMAIL_TEMPLATE"     envDefault:"email.tmpl"`
+	Host                   string `env:"SMQ_EMAIL_HOST"                      envDefault:"localhost"`
+	Port                   string `env:"SMQ_EMAIL_PORT"                      envDefault:"25"`
+	Username               string `env:"SMQ_EMAIL_USERNAME"                  envDefault:"root"`
+	Password               string `env:"SMQ_EMAIL_PASSWORD"                  envDefault:""`
+	FromAddress            string `env:"SMQ_EMAIL_FROM_ADDRESS"              envDefault:""`
+	FromName               string `env:"SMQ_EMAIL_FROM_NAME"                 envDefault:""`
+	Template               string `env:"SMQ_EMAIL_TEMPLATE"                  envDefault:"email.tmpl"`
+	PasswordResetURLPrefix string `env:"SMQ_USERS_PASSWORD_RESET_URL_PREFIX" envDefault:"http://localhost:8080"`
```

Comment:
Don't pollute `internal/email` with this. Email has nothing to do with password reset; it only send whatever email you get him send. Rather, do something like (i.e. what we already have, but don't pass host param to the `SendPasswordReset`:

```suggestion

type emailer struct {
	resetURL string
	agent    *email.Agent
}

// New creates new emailer utility.
func New(resetURL string, c *email.Config) (users.Emailer, error) {
	e, err := email.New(c)
	return &emailer{
		resetURL: resetURL,
		agent:    e,
	}, err
}

func (e *emailer) SendPasswordReset(to []string, user, token string) error {
	cont := fmt.Sprintf("%s?token=%s", e.resetURL, token)
	return e.agent.Send(to, "", "Password Reset Request", "", user, url, "")
}

```